### PR TITLE
Enhance scheduler documentation

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area documentation
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR enhances the Gardener scheduler documentation by an algorithm overview to let operators easily understand the logic of determine seed candidates.

**Special notes for your reviewer**:
/invite @danielfoehrKn 

**Release note**:

```doc operator
Gardener's scheduler documentation has been enhanced. It concisely explains the algorithm used to determine seed candidates.
```

### CODE REVIEW

File: docs/concepts/scheduler.md [hunk 1]
```diff
@@ -20,6 +20,21 @@ It should be possible to easily extend and tweak the scheduler in the future.
...
+1. Determine usable seeds (non-deleting, visible)
+1. Consider seeds matching `.spec.seedSelector` in `CloudProfile` used by the `Shoot`
+1. Consider seeds matching `.spec.seedSelector` in `Shoot`
+1. Filter seeds:
+   * that have no network intersection with the `Shoot`'s networks (due to the VPN connectivity between seeds and shoots their networks must be disjoint)
+   * that have `.spec.settings.shootDNS.enabled=false` (only if the shoot specifies a DNS domain or does not use the `unmanaged` DNS provider)
+   * whose taints (`.spec.taints`) are tolerated by the `Shoot` (`.spec.tolerations`)
+   * whose capacity for shoots would be exceeded if the shoot is scheduled onto the seed, see [Ensuring seeds capacity for shoots is not exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded)
+1. Apply active [strategy](#strategies) e.g., _Minimal Distance strategy_
+1. Choose least utilized seed, i.e., the one with the least number of shoot control planes, will be the winner and written to the `.spec.seedName` field of the `Shoot`.
```
Comment (rfranzke) [inline docs/concepts/scheduler.md]: You can also mention that the `Seed` must be "Ready", i.e., the `Bootstrapped` condition must be `True`.

Comment (timuthy) [reply docs/concepts/scheduler.md]: Good idea, thanks.

File: docs/concepts/scheduler.md [hunk 2]
```diff
@@ -20,6 +20,24 @@ It should be possible to easily extend and tweak the scheduler in the future.
 Possibly, similar to the Kubernetes scheduler, hooks could be provided which influence the scheduling decisions.
 It should be also possible to completely replace the standard Gardener Scheduler with a custom implementation.
 
+## Algorithm overview
+
+The following **sequence** describes the steps involved to determine a seed candidate:
+
+1. Determine usable seeds
+   * no `.metadata.deletionTimestamp`
+   * visible `.spec.settings.scheduling.visible` is `true`
```
Comment (rfranzke) [inline docs/concepts/scheduler.md]:
```suggestion
   * `.spec.settings.scheduling.visible` is `true`
```

File: docs/concepts/scheduler.md [hunk 3]
```diff
@@ -20,6 +20,24 @@ It should be possible to easily extend and tweak the scheduler in the future.
 Possibly, similar to the Kubernetes scheduler, hooks could be provided which influence the scheduling decisions.
 It should be also possible to completely replace the standard Gardener Scheduler with a custom implementation.
 
+## Algorithm overview
+
+The following **sequence** describes the steps involved to determine a seed candidate:
+
+1. Determine usable seeds
```
Comment (rfranzke) [inline docs/concepts/scheduler.md]:
```suggestion
1. Determine usable seeds with "usable" defined as follows:
```

File: docs/concepts/scheduler.md [hunk 4]
```diff
@@ -20,6 +20,24 @@ It should be possible to easily extend and tweak the scheduler in the future.
...
+   * no `.metadata.deletionTimestamp`
+   * visible `.spec.settings.scheduling.visible` is `true`
+   * conditions `Bootstrapped`, `GardenletReady` are `true`
+1. Consider seeds matching `.spec.seedSelector` in `CloudProfile` used by the `Shoot`
+1. Consider seeds matching `.spec.seedSelector` in `Shoot`
```
Comment (rfranzke) [inline docs/concepts/scheduler.md]: Can this be put into the "Filter seeds" block as well?

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: docs/concepts/scheduler.md [hunk 5]
```diff
@@ -20,6 +20,24 @@ It should be possible to easily extend and tweak the scheduler in the future.
...
+   * no `.metadata.deletionTimestamp`
+   * `.spec.settings.scheduling.visible` is `true`
+   * conditions `Bootstrapped`, `GardenletReady` are `true`
+1. Filter seeds:
+   * matching `.spec.seedSelector` in `CloudProfile` used by the `Shoot`
+   * matching `.spec.seedSelector` in `Shoot`
+   * having no network intersection with the `Shoot`'s networks (due to the VPN connectivity between seeds and shoots their networks must be disjoint)
+   * having `.spec.settings.shootDNS.enabled=false` (only if the shoot specifies a DNS domain or does not use the `unmanaged` DNS provider)
+   * whose taints (`.spec.taints`) are tolerated by the `Shoot` (`.spec.tolerations`)
+   * whose capacity for shoots would be exceeded if the shoot is scheduled onto the seed, see [Ensuring seeds capacity for shoots is not exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded)
```
Comment (dguendisch) [inline docs/concepts/scheduler.md]:
```suggestion
   * whose capacity for shoots would not be exceeded if the shoot is scheduled onto the seed, see [Ensuring seeds capacity for shoots is not exceeded](#ensuring-seeds-capacity-for-shoots-is-not-exceeded)
```

**Review by dguendisch — CHANGES_REQUESTED** [review]

File: docs/concepts/scheduler.md (hunk 5 above)
Comment (timuthy) [reply docs/concepts/scheduler.md]: Nice catch, thanks @dguendisch

**Review by dguendisch — APPROVED** [review]
Thanks :)

**Review by danielfoehrKn — APPROVED** [review]
Looks great!  Thanks

**Pull request merged**